### DISCUSSION TITLE
Avatar Group API RFC

### DISCUSSION THREAD

**Discussion by nkrantz** [twilio-labs/paste#3153] [Review request]
## Hello from Paste

Thanks for taking the time to provide some feedback to Paste! We appreciate your contribution.

Please take a look at the following API proposal for the new Avatar Group component. Also sometimes known as Face Pile or Stacked Connections, Avatar Group displays multiple Avatars together to represent people or entities within the same group. Avatar Group will be exported from the same package as the existing [Avatar component](https://paste.twilio.design/components/avatar).

<img width="105" alt="image" src="https://user-images.githubusercontent.com/75342690/233192207-a38cc489-b7f1-47f1-a76c-ee2a6fc29b49.png">

Proposed API:

```
<AvatarGroup size="sizeIcon90">
  <Avatar size="sizeIcon90" name="Nora Krantz"/>
  <Avatar size="sizeIcon90" name="Loreina Chew"/>
  <Avatar size="sizeIcon90" name="Shadi Isber"/>
</AvatarGroup>
```

Note: The proposed API has a required prop, `size`, which should match the token value passed to the Avatar components. The spacing of the Avatar Group is determined based on this value. All Avatars in an Avatar Group should be the same size.

[Check out the API Exploration CodeSandbox](https://codesandbox.io/s/avatar-group-api-exploration-3twid1).

Please share any feedback or suggestions you might have! Thanks.

**Comment by TheSisb** [comment]
Hi @nkrantz,

Thanks for piloting the RFC process! 👏 

Giving this some more thought, maybe we should use context to mimic our other components after all.  We can pass `size` as it is in your example to AvatarGroup, which will set it on a context provider. Then, we can modify Avatar to get size from the context consumer.

Example:

```js
<AvatarGroup size="sizeIcon90">
  <Avatar name="Nora Krantz"/>
  <Avatar name="Loreina Chew"/>
  <Avatar name="Shadi Isber"/>
</AvatarGroup>
```

Added benefit, since we don't want odd sized avatars in a group, we can make the context take precedence. So that, in the following example, sizeIzon50 is ignored:

```js
<AvatarGroup size="sizeIcon90">
  <Avatar name="Nora Krantz"/>
  <Avatar size="sizeIcon50" name="Loreina Chew"/>
  <Avatar name="Shadi Isber"/>
</AvatarGroup>
```


It's a small improvement over spreading it everywhere and should be easy to do.

**Comment by TheSisb** [comment]
The component has been released and the window for comment has been closed.
